tests/robot-tests/tests/libs/common.robot

```diff
@@ -1014,3 +1013,20 @@ user takes html snapshot of element
 
 user waits for caches to expire
     sleep    %{WAIT_CACHE_EXPIRY}
+
+user wait for option to be available and select it
+    [Arguments]  ${dropdown_locator}  ${option_text}  ${timeout}=%{TIMEOUT}
+    wait until keyword succeeds  ${timeout}  1s  check option exist in dropdown  ${dropdown_locator}  ${option_text}
+    select from list by label  ${dropdown_locator}  ${option_text}
+
+check option exist in dropdown
+    [Arguments]  ${dropdown_locator}  ${option_text}
+    ${options}=  get webelements  ${dropdown_locator} > option
+    ${matched}=  set variable  False
+
+    FOR  ${option}  IN  @{options}
+    ${text}=  get text  ${option}
+    run keyword if  '${text}' == '${option_text}'  Set Variable  ${matched}  True
+    should be true  ${matched}  Option ${option_text} not found in the dropdown
+    END
```

Comment:
I think this for loop isn't going to work. The final line in the for loop needs to be outside the loop, as right now this loops expects *every* element in `options` to match with `${option_text}`.

```
    FOR  ${option}  IN  @{options}
    ${text}=  get text  ${option}
    run keyword if  '${text}' == '${option_text}'  Set Variable  ${matched}  True
    END

    should be true  ${matched}  Option ${option_text} not found in the dropdown
```

_Hopefully_ that'll do it.

Reply:
I managed tried the way that you have mentioned, it looks like its doing linear search (checking each element against set of element). Due to which its failing. May be I am working on different approach, seems promising . Once its done, I will push latest changes